### ISSUE TITLE
Remove alt -> ctrl+arrow hack in favor of embedder-specific solutions

### ISSUE THREAD

**Issue by Tyriar** [xtermjs/xterm.js#4538]
Context: https://github.com/xtermjs/xterm.js/issues/264#issuecomment-1505538266

Finding the commit they were added to confirm is also a good idea.

https://github.com/xtermjs/xterm.js/blob/8b422442b7079525dca8b2a1c6b167856d1d7e7e/src/common/input/Keyboard.ts#L114-L189

**Comment by fifv** [comment]
Thank you [Tyriar](https://github.com/Tyriar) for saving my life. I have spent hours to figure out why my Alt key not works...

For anyone who wants a workaround, here is my hack for tabby (other terminals should be similar):

1. Open `C:\Users\<USER>\AppData\Local\Programs\Tabby\resources\builtin-plugins\tabby-terminal\dist\index.js`
    - This is xterm.js ‘s problem, and Tabby use xterm.js as a plugin, note it belongs to `builtin-plugins` instead of main tabby app
   
2. Fix for Arrows:
    
    Find `[1;5D` (may ~27452 at Tabby 1.0.196)
    
    **Remove** these selected
    
    ![image](https://github.com/xtermjs/xterm.js/assets/75289510/08e4c877-bdce-4ba0-835d-d2f017d026ca)

    
3. Fix for PageUp/PageDown:
    
    Find `[5~` (may ~27479 at Tabby 1.0.196, just near the above ones)
    
    **Add** these selected 
    
    ![image](https://github.com/xtermjs/xterm.js/assets/75289510/c94caf9c-2a00-46cd-8b2e-5f03a976a686)

**Comment by dmarcuse** [comment]
Is there any recent update on this? I've been frustrated by this behavior in VS code and finally figured out this was the cause (and found the linked workaround via VS code's shortcuts). I see the help wanted tag - if it's just a matter of getting someone to do it, I'm happy to make the changes.

**Comment by dmarcuse** [comment]
Sounds good, I'll set aside some time to work on this over the weekend then.

**Comment by balthild** [comment]
You could just "fix" it by simply override the key binding if you're using vscode. Maybe the `workbench.action.terminal.sendSequence` action does not send sequences via xterm.js, but through conpty directly, so it would not be modified?

```jsonc

    {
        "key": "alt+left",
        "command": "workbench.action.terminal.sendSequence",
        "args": {
            "text": "\u001b[1;3D"
        },
        "when": "terminalFocus"
    },
    {
        "key": "alt+right",
        "command": "workbench.action.terminal.sendSequence",
        "args": {
            "text": "\u001b[1;3C"
        },
        "when": "terminalFocus"
    },
```

Result:

![image](https://github.com/user-attachments/assets/dd106d09-357e-4562-a48a-74906a7d22a2)

![image](https://github.com/user-attachments/assets/d695501e-dde3-4e66-824e-faf5256f798e)

**Comment by Tyriar** [comment]
@balthild yep, we just need that keybinding built in to VS Code. So those keybindings added somewhere over here: https://github.com/microsoft/vscode/blob/3ae6c5ef651585b12870828230f214eed2e6ef07/src/vs/workbench/contrib/terminal/browser/terminal.contribution.ts#L245-L249

And then remove the overrides from xterm.js

**Comment by jpmorris** [comment]
> You could just "fix" it by simply override the key binding if you're using vscode. Maybe the 

Worked for me. Note: you cant have  'Send Keybindings To Shell' enabled:

https://github.com/microsoft/vscode/issues/190629#issuecomment-1682733387
